sdk/search/azure-search-documents/azure/search/documents/indexes/models/_index.py

```diff
@@ -31,26 +31,26 @@ class SearchField(msrest.serialization.Model):
 
     All required parameters must be populated in order to send to Azure.
 
-    :param name: Required. The name of the field, which must be unique within the fields collection
+    :keyword name: Required. The name of the field, which must be unique within the fields collection
```

Comment:
I think we should use "param" rather than "keyword" for required parameters?

Reply:
From guideline:

```py
def get(*args, **kwargs):
    """Calls `request` with the method "GET" and forwards all other arguments.

    :param str method-param: The method-param parameter
    :keyword int method-kwarg: The optional method-kwarg parameter

    For additional request configuration options, please see https://aka.ms/azsdk/python/options.
    """
    return request("GET", *args, **kwargs)
```


https://azure.github.io/azure-sdk/python_documentation.html

Reply:
https://github.com/Azure/azure-sdk-for-python/blob/main/doc/dev/docstring_typehint.md#model-and-client-docstrings

I don't think the distinction between `:param` and `:keyword` is whether the argument is required or not, but rather whether it is positional or keyword-only, and in this case (all cases in this PR) they are keyword-only.

@johanste @annatisch can you comment here?

If we leave these as `:param`, APIView will _not_ display them.

Reply:
Correct -  there is nothing that prevents keyword-only required parameters (although we generally try to avoid them in convenience layers if possible)